# HG changeset patch
# Parent 86b226c5b3b486d40efc6095689e0b2dd1c38fdd
Stop being retarded on includes

Fix build warnings on apple/clang.

diff --git a/tests/core/CMakeLists.txt b/tests/core/CMakeLists.txt
--- a/tests/core/CMakeLists.txt
+++ b/tests/core/CMakeLists.txt
@@ -1,1 +1,10 @@
+INCLUDE_DIRECTORIES(
+    ${LIGHTSD_SOURCE_DIR}
+    ${LIGHTSD_SOURCE_DIR}/core/
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${LIGHTSD_BINARY_DIR}
+    ${LIGHTSD_BINARY_DIR}/core/
+    ${CMAKE_CURRENT_BINARY_DIR}
+)
+
 ADD_ALL_SUBDIRECTORIES()
diff --git a/tests/core/jsonrpc/CMakeLists.txt b/tests/core/jsonrpc/CMakeLists.txt
--- a/tests/core/jsonrpc/CMakeLists.txt
+++ b/tests/core/jsonrpc/CMakeLists.txt
@@ -1,10 +1,6 @@
 INCLUDE_DIRECTORIES(
-    ${LIGHTSD_SOURCE_DIR}
-    ${LIGHTSD_SOURCE_DIR}/../
-    ${LIGHTSD_SOURCE_DIR}/core/
-    ${LIGHTSD_BINARY_DIR}
-    ${LIGHTSD_BINARY_DIR}/../
-    ${LIGHTSD_BINARY_DIR}/core/
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR}
 )
 
 ADD_LIBRARY(
diff --git a/tests/core/router/CMakeLists.txt b/tests/core/router/CMakeLists.txt
--- a/tests/core/router/CMakeLists.txt
+++ b/tests/core/router/CMakeLists.txt
@@ -1,12 +1,6 @@
 INCLUDE_DIRECTORIES(
-    ${LIGHTSD_SOURCE_DIR}
-    ${LIGHTSD_SOURCE_DIR}/../
-    ${LIGHTSD_SOURCE_DIR}/core/
-    ${LIGHTSD_SOURCE_DIR}/tests/core/
-    ${LIGHTSD_BINARY_DIR}
-    ${LIGHTSD_BINARY_DIR}/../
-    ${LIGHTSD_BINARY_DIR}/core/
-    ${LIGHTSD_BINARY_DIR}/tests/core/
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR}
 )
 
 ADD_LIBRARY(
diff --git a/tests/core/router/tests_router_utils.h b/tests/core/router/tests_router_utils.h
--- a/tests/core/router/tests_router_utils.h
+++ b/tests/core/router/tests_router_utils.h
@@ -1,20 +1,3 @@
-#include <sys/queue.h>
-#include <sys/tree.h>
-#include <sys/socket.h>
-#include <netinet/in.h>
-#include <stdbool.h>
-#include <stdlib.h>
-#include <stdint.h>
-#include <string.h>
-
-#include <event2/util.h>
-
-#include "lifx/wire_proto.h"
-#include "core/time_monotonic.h"
-#include "lifx/bulb.h"
-#include "lifx/gateway.h"
-#include "tests_utils.h"
-
 int lgtd_tests_gw_pkt_queue_size = 0;
 struct {
     struct lgtd_lifx_gateway        *gw;
